server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -833,14 +834,26 @@ public Symbol visitArrayLikePredicate(ArrayLikePredicate node, ExpressionAnalysi
 
         @Override
         protected Symbol visitLikePredicate(LikePredicate node, ExpressionAnalysisContext context) {
-            if (node.getEscape() != null) {
-                throw new UnsupportedOperationException("ESCAPE is not supported.");
-            }
             Symbol expression = node.getValue().accept(this, context);
             Symbol pattern = node.getPattern().accept(this, context);
+            Symbol escape;
+            String escapeValue;
+            if (node.getEscape() != null) {
+                escape = node.getEscape().accept(this, context);
+                // Aligned with PG: an empty ESCAPE falls back to default backslash.
+                assert escape instanceof Literal : "ESCAPE must be provided as a literal.";
+                escapeValue = (String) ((Literal<?>) escape).value();
+                if (escapeValue.isEmpty()) {
+                    escape = Literal.of(String.valueOf(DEFAULT_ESCAPE));
```

Comment:
Imho, this should not be done in the parser, logically, how to treat the missing optional `ESCAPE`, should be done in the operators.

Reply:
Done in https://github.com/crate/crate/pull/15166/commits/bfb0023e3eb4087962a711b588eea0fa4c25da40
As discussed, added new signature because of BWC